_posts/wiki/2019-09-19-Franka-Installation-Tutorial.md

```diff
@@ -0,0 +1,142 @@
+curl -SLO https://www.kernel.org/pub/linux/kernel/projects/rt/<your version number ex: 4.12 >/older/patch-<your version>-rt10.patch.sign
+~~~
+
+Unpack Patch:
+
+~~~bash
+xz -d linux-<your version>.tar.xz
+xz -d patch-<your version>-rt10.patch.xz
+~~~
+
+# Compiling the Kernel
+
+Extract the Source Code:
+
+ ~~~bash
+ tar xf linux-<your version>.tar
+ cd linux-<your version>
+ patch -p1 < ../patch-<your version>-rt10.patch
+ ~~~
+
+Configure Kernel:
+
+ ~~~bash
+ make oldconfig
+ When menu pops up choose option 5: Fully Preemptible Kernel
+ # For the rest of the options, use default values
+ # if fakeroot doesn't work, use sudo make deb-pkg
+ fakeroot make deb-pkg
+ sudo dpkg -i ../linux-headers-<your version>-rt10_*.deb ../linux-image-<your version>-rt10_*.deb
+ ~~~
+
+# Verifying the Kernel
+
+Restart the computer, go to the Grub boot menu, select your kernel. After entering `uname -a` you should see the output which contains `PREEMPT_RT`.
+realtime should exist at `/sys/kernel/realtime` and contain the number `1`.
+
+# Adding Permissions
+
+After kernel is running you need to create then add the user controlling the robot to the group named `realtime`:
+
+ ~~~bash
+ sudo addgroup realtime
+ sudo usermod -a -G realtime <username here>
+ ~~~
+
+Add the following to the realtime group in `/etc/security/limits.conf` then logout and log back in:
+
+ ~~~bash
+ @realtime soft rtprio 99
+ @realtime soft priority 99
+ @realtime soft memlock 102400
+ @realtime hard rtprio 99
+ @realtime hard priority 99
+ @realtime hard memlock 102400
+ ~~~
+
+# Connecting to the Panda
+
+* Go to your router's website, login with your user-name and password and find the Panda's IP address. Enter the Panda's IP to the address bar on a web browser and log in to the Franka Desk using your user-name and password or if it's the first time you are connecting to the Franka Desk the user-name and password are created when you enter them.
+* To use a static IP be sure that you have a direct connection to the Panda: login to the Franka Desk, navigate to the network settings, uncheck the DHCP option, enter the Panda's IP in the address field and then enter the network mask you wish to use into the network field. Configure the workstation's network by going to the edit connections in the network settings, clicking edit on the connection you wish to use, switching it to manual, and then entering the address of the Panda and the network mask you are using. Be sure to switch back to automatic when you are finished using the Franka control interface.
+* To use the DHCP simply enter the Panda's IP in your browser and login to the Franka Desk using your user-name and password.
+
+# Using the Panda
+
+ * Unlock the joints from the control interface before using the Panda.
+ * When shutting the Panda down be sure to wait for the robot to completely shut down before exiting the interface window.
+ * Create great things!
```

Comment:
Can you change this last line to "Go do good stuff!" please? There is no reason in particular but it's the sentence we used in my previous lab and since you were so close I'd like to rather use that :-)

Reply:
Sure no problem!